superset-frontend/src/utils/copy.ts

```diff
@@ -17,40 +17,71 @@
  * under the License.
  */
 
-const copyTextToClipboard = async (text: string) =>
-  new Promise<void>((resolve, reject) => {
-    const selection: Selection | null = document.getSelection();
-    if (selection) {
-      selection.removeAllRanges();
-      const range = document.createRange();
-      const span = document.createElement('span');
-      span.textContent = text;
-      span.style.position = 'fixed';
-      span.style.top = '0';
-      span.style.clip = 'rect(0, 0, 0, 0)';
-      span.style.whiteSpace = 'pre';
+// Use the new Clipboard API if the browser supports it
+const copyTextWithClipboardApi = async (getText: () => Promise<string>) => {
+  try {
+    // Safari (WebKit) does not support delayed generation of clipboard.
+    // This means that writing to the clipboard, from the moment the user
+    // interacts with the app, must be instantaneous.
+    // However, neither writeText nor write accepts a Promise, so
+    // we need to create a ClipboardItem that accepts said Promise to
+    // delay the text generation, as needed.
+    // Source: https://bugs.webkit.org/show_bug.cgi?id=222262
+    const clipboardItem = new ClipboardItem({
+      'text/plain': getText(),
+    });
+    await navigator.clipboard.write([clipboardItem]);
+  } catch {
+    // For Blink, the above method won't work, but we can use the
+    // default (intended) API, since the delayed generation of the
+    // clipboard is now supported.
+    // Source: https://bugs.chromium.org/p/chromium/issues/detail?id=1014310
+    const text = await getText();
+    await navigator.clipboard.writeText(text);
+  }
+};
 
-      document.body.appendChild(span);
-      range.selectNode(span);
-      selection.addRange(range);
+const copyTextToClipboard = (getText: () => Promise<string>) =>
+  copyTextWithClipboardApi(getText)
+    // If the Clipboard API is not supported, fallback to the older method.
```

Comment:
nit: it could clean up the code by breaking the old logic into a separate function